<?xml version="1.0"?>

<launch>
	
	<!--node pkg="centauro_tools" 
          type="centauro_cop_estimation" 
          name="centauro_cop_estimation"
          output="screen">
		  
          <param name="rate" value="200.0"/>
		  
	</node-->

  <!-- Mine additions -->
  <!-- centauro configuration parameters -->
  <arg name="arms" default="true"/>
  <arg name="legs" default="true"/>

<!--param name="robot_description"
         command="$(find xacro)/xacro --inorder '$(find centauro_urdf)/urdf/centauro_robot.urdf.xacro' add_upper_body:=$(arg arms) add_legs:=$(arg legs)"/-->
	<param name="robot_description_semantic" textfile="$(find centauro_srdf)/srdf/centauro.srdf"/>

    <node pkg="cartesian_interface" 
          type="force_estimation_node" 
          name="force_estimator"
          output="screen">
		  
		<remap from="~torque_offset" to="cartesian/torque_offset"/>
	
        <param name="is_model_floating_base" value="true"/>
        <param name="model_type" value="RBDL"/>
        
        <!--param name="links" type="yaml" value="[ankle2_1, ankle2_2, ankle2_3, ankle2_4]"/-->
        <param name="links" type="yaml" value="[contact_1, contact_2, contact_3, contact_4]"/>
        <!--param name="links" type="yaml" value="[wheel_1, wheel_2, wheel_3, wheel_4]"/-->

        <param name="chains" type="yaml" value="[front_left_leg, front_right_leg, rear_right_leg, rear_left_leg]"/>
        
        <param name="arm1_8/dofs" type="yaml" value="[0, 1, 2]"/>
        <param name="arm2_8/dofs" type="yaml" value="[0, 1, 2]"/>

        <param name="contact_1/dofs" type="yaml" value="[0, 1, 2]"/>
        <param name="contact_2/dofs" type="yaml" value="[0, 1, 2]"/>
	    <param name="contact_3/dofs" type="yaml" value="[0, 1, 2]"/>
        <param name="contact_4/dofs" type="yaml" value="[0, 1, 2]"/>

	    <!--param name="ankle2_1/dofs" type="yaml" value="[0, 1, 2]"/>
        <param name="ankle2_2/dofs" type="yaml" value="[0, 1, 2]"/>
	    <param name="ankle2_3/dofs" type="yaml" value="[0, 1, 2]"/>
        <param name="ankle2_4/dofs" type="yaml" value="[0, 1, 2]"/-->

        <!--param name="wheel_1/dofs" type="yaml" value="[0, 1, 2]"/>
        <param name="wheel_2/dofs" type="yaml" value="[0, 1, 2]"/>
	    <param name="wheel_3/dofs" type="yaml" value="[0, 1, 2]"/>
        <param name="wheel_4/dofs" type="yaml" value="[0, 1, 2]"/-->

		<param name="rate" value="800.0"/>
        
    </node>

    <node name="feet_contacts_node" pkg="centauro_contact_detection" type="contact_detection_time_window" output="screen">
        <param name="pub_freq" value="800"/>
        <param name="force_thold" value="100"/>
        <!--param name="window" value="$(arg time_window)"/-->
    </node>

</launch>